app/models/service_request.rb

```diff
@@ -566,6 +560,16 @@ def is_editable?
     return true
   end
 
+  def has_locked_ssr?
+    sub_service_requests.each do |ssr|
+      if !ssr.can_be_edited?
+        return true
```

Comment:
I think we could, but by using the early return, we can potentially save time by not going through all SSRs. A single one which can't be edited means the SR should show, so we shouldn't have to continue iterating at that point